src/julia.grammar

```diff
@@ -136,7 +136,7 @@ definition[@isGroup=Definition] {
   | PrimitiveDefinition { kwid<'primitive'> kwid<'type'> TypeHead IntegerLiteral end }
   | StructDefinition { kwid<'mutable'>? kw<'struct'> TypeHead (_t block?)? end } // TODO: Allow skipping terminator
   | ModuleDefinition { (kw<'module'> | kw<'baremodule'>) Identifier _t? block? end }
-  | MacroDefinition { kw<'macro'> Signature { CallExpression } _t? block? end }
+  | MacroDefinition { kw<'macro'> ( Signature { Identifier } | Signature { CallExpression } ) _t? block? end }
```

Comment:
Functions have a separate case for identifiers because zero method functions cannot have bodies.

It might be better to have: 

```suggestion
  | MacroDefinition { kw<'macro'> (Signature { Identifier } | (Signature { CallExpression } _t? block? end)) }
```